src/split-criterions/GiniImpurity.jl

```diff
@@ -77,33 +75,33 @@ Args:
 Returns:
     A tuple containing the index of the best feature and the best split value.
 """
-function find_best_split(data::Matrix{T}, labels::Vector{L}, num_features_to_use::Int=-1) where {T, L}  
-    best_gini = Inf  
-    best_feature_index = -1  
-    best_feature_value = -1  
+function find_best_split(data::AbstractMatrix, labels::AbstractVector, num_features_to_use::Int=-1)
+    best_gini = Inf
+    best_feature_index = -1
```

Comment:
we should introduce consistent variable names in our splitting functions. For example, in gini impurity and information gain the variables "best_feature_index" and "best_feature" have the same functionality.

Reply:
i set the default feature index values to -1 for all of the splitting criterions and changed it in the build_tree function as well